pkgs/os-specific/linux/kernel/linux-mptcp.nix

```diff
@@ -43,4 +43,4 @@ import ./generic.nix (args // rec {
     TCP_CONG_BALIA m
 
   '' + (args.extraConfig or "");
-} // (args.argsOverride or {}))
+} // args // (args.argsOverride or {}))
```

Comment:
It looks like this code was designed to use argsOverride to do what you are doing, not args directly.
In this case, your override should look like

```nix
  pkg.override ({
    argsOverride = {
      modDirVersion="4.9.60+";
      src=pkgs.lib.cleanSource /home/teto/mptcp;
    };
  })
```

Now, this may not be the most obvious way to do it. However, I would prefer to keep this consistent with the other kernel implementations.

Reply:
I saw the args.argsOverride afterwards. Any technical reason for args.argsOverride or is it legacy ? seeing that there is no specific documentation for kernels, people are bound to look at the nix code either way or use override out of habit. 
Btw I would be interested in your opinon over https://github.com/NixOS/nixpkgs/pull/31463 :)

Reply:
Really no idea... You could get more info by blame'ing on original kernels.

Reply:
well so far it works but I had lots of trouble compiling the kernel for many reasons; i'll try to send some doc for the manual.